### PR TITLE
Add HIDcomposite class and example for multimedia keyboards

### PR DESCRIPTION
**Pull request by HakanL**
Not sure this is the cleanest way, please comment.

### CODE REVIEW

File: examples/HID/USBHIDMultimediaKbd/USBHIDMultimediaKbd.ino [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+protected:
+  void ParseHIDData(USBHID *hid, uint8_t ep, bool is_rpt_id, uint8_t len, uint8_t *buf); // Called by the HIDComposite library
+  bool SelectInterface(uint8_t iface, uint8_t proto);
+};
+
+bool HIDSelector::SelectInterface(uint8_t iface, uint8_t proto)
+{
+  if(proto != 0)
+    return true;
```
Comment (Lauszus) [inline examples/HID/USBHIDMultimediaKbd/USBHIDMultimediaKbd.ino]: Can you add a comment here, so newcomers will understand what is going on?

**Comment by Lauszus** [comment]
Thanks for the PR. I have made one inline comment. Also there seems to be some indentation issues in both the .h and .cpp file. It properly is due to a mix of space and tabs for whitespaces? I would recommend using something like [Sublime Text](https://www.sublimetext.com) to fix it.

**Comment by HakanL** [comment]
Should be fixed now.

**Comment by Lauszus** [comment]
@HakanL it looks like you got the commit 8336ddd7331aa4f0a54c8f62b31189297331658a merged into this PR somehow?

**Comment by HakanL** [comment]
Yes, that's usually how I've seen changes to PRs been done. So you don't have to have multiple PRs.

**Comment by Lauszus** [comment]
@HakanL no normally unrelated commits go into a separate PR, as it was already done: https://github.com/felis/USB_Host_Shield_2.0/pull/220. If you have problems fixing it. I will just merge it manually.

**Comment by HakanL** [comment]
Since this PR hasn't been merge in I don't seem to be able to create a new PR, GitHub just assumes that if you add more commits to the branch that the PR was tied to, that it goes against that PR. Here's the commit for the changes: https://github.com/HakanL/USB_Host_Shield_2.0/commit/83a52b5063c0e051cbdb7a64b97d39c2a0c53d4b

**Comment by Lauszus** [comment]
@HakanL if you want to create another PR you will have to create a new branch and then open a new one.

**Comment by HakanL** [comment]
ok, I've never seen any OSS do it like this on Github, but no worries, I've created another branch and PR.

**Comment by Lauszus** [comment]
@HakanL this is how Github works and is not specific to this project.

Anyway I will merge your PR now. Thanks for your contribution!

**Comment by HakanL** [comment]
On other projects I've made iterative changes to the same PR based on comments until it's been merged/accepted, but no big deal, we got the code in there, that's what counts :+1:

**Pull request merged**